crates/scarb-artifacts/src/lib.rs

```diff
@@ -84,25 +84,34 @@ pub fn try_get_starknet_artifacts_path(
 }
 
 /// Get the map with `StarknetContractArtifacts` for the given package
-///
-/// # Arguments
-///
-/// * `path` - A path to `{target_name}.starknet_artifacts.json` file
-pub fn get_contracts_map(path: &Utf8Path) -> Result<HashMap<String, StarknetContractArtifacts>> {
-    let base_path = path
-        .parent()
-        .ok_or_else(|| anyhow!("Failed to get parent for path = {}", path))?;
-    let artifacts = artifacts_for_package(path)?;
-    let mut map = HashMap::new();
-    for contract in artifacts.contracts {
-        let name = contract.contract_name;
-        let sierra_path = base_path.join(contract.artifacts.sierra);
-        let casm_path = base_path.join(contract.artifacts.casm);
-        let sierra = fs::read_to_string(sierra_path)?;
-        let casm = fs::read_to_string(casm_path)?;
-        map.insert(name, StarknetContractArtifacts { sierra, casm });
+pub fn get_contracts_map(
+    metadata: &Metadata,
+    package: &PackageId,
+) -> Result<HashMap<String, StarknetContractArtifacts>> {
+    let target_dir = target_dir_for_package(metadata);
+    let target_name = target_name_for_package(metadata, package)?;
+
+    let maybe_contracts_path = try_get_starknet_artifacts_path(&target_dir, &target_name)?;
+
+    match maybe_contracts_path {
+        Some(contracts_path) => {
+            let base_path = contracts_path
+                .parent()
+                .ok_or_else(|| anyhow!("Failed to get parent for path = {}", &contracts_path))?;
+            let artifacts = artifacts_for_package(&contracts_path)?;
+            let mut map = HashMap::new();
+            for contract in artifacts.contracts {
+                let name = contract.contract_name;
+                let sierra_path = base_path.join(contract.artifacts.sierra);
+                let casm_path = base_path.join(contract.artifacts.casm);
+                let sierra = fs::read_to_string(sierra_path)?;
+                let casm = fs::read_to_string(casm_path)?;
+                map.insert(name, StarknetContractArtifacts { sierra, casm });
```

Comment:
Please add method `StarknetContractArtifacts::from_scarb_contract_artifact(StarknetContract)` I think it is the only way we read it, we can abstract this away.

Please move the whole logic under this pattern to `load_contract_artifacts` function, right now you need to read the whole code to understand what it does and cognitive complexity is high.